graphql/schemabuilder/pagination.go

```diff
@@ -433,65 +433,103 @@ func (c *connectionContext) applyBatchTextFilter(ctx context.Context, nodes []in
 	return nodesToKeep, nil
 }
 
+func (c *connectionContext) checkFilters(ctx context.Context, node interface{}, matchStrings []string, filterFields map[string]*graphql.Field) (bool, error) {
+	keep := false
+	for name, filterField := range filterFields {
+		// Resolve the graphql.Field made for sorting.
+		text, err := filterField.Resolve(ctx, node, nil, nil)
+		if err != nil {
+			return keep, err
+		}
+		// Only strings are allowed for FilterText fields.
+		textString, ok := text.(string)
+		if !ok {
+			return keep, fmt.Errorf("filter %s returned %T, must be a string", name, text)
+		}
+		if filter.MatchText(textString, matchStrings) {
+			keep = true
+			break
+		}
+	}
+	return keep, nil
+}
+
+func (c *connectionContext) applyTextFilterNotBatchedExpensive(ctx context.Context, nodes []interface{}, matchStrings []string, filterFields map[string]*graphql.Field) ([]bool, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	nodesToKeep := make([]bool, len(nodes))
```

Comment:
I noticed that we're using a mutex in `applyBatchTextFilter`, but not in this function.  I think maybe the mutex is leftover from when we were using a map rather than a slice?

Reply:
for batching you need the mutex since we make multiple changes to the same index in nodesToKeep, but for both applyTextFilterNotBatchedExpensive and applyTextFilterNotBatched we only touch each index once, so we don't need the mutex lock

Reply:
Ah, that makes sense.  Thanks!